src/jit/emitxarch.cpp

```diff
@@ -9185,6 +9191,26 @@ BYTE* emitter::emitOutputRR(BYTE* dst, instrDesc* id)
 
 #endif // _TARGET_AMD64_
     }
+#if defined(_TARGET_XARCH_) && !defined(LEGACY_BACKEND)
+    else if ((ins == INS_crc32) || (ins == INS_lzcnt) || (ins == INS_popcnt))
+    {
+        code = insEncodeRMreg(ins, code);
+        if (ins == INS_crc32 && size > EA_1BYTE)
+        {
+            code |= 0x0100;
+        }
+
+        if (size == EA_2BYTE)
```

Comment:
Do we really need to support 16 bit operands?

Reply:
Yes, these three intrinsics all need.

Reply:
Hmm, just because 16 bit versions of these instructions exist? This may be problematic because these instructions produce 16 bit values, you may need to add a `movzx` after such instructions to keep the JIT happy.

Reply:
> This may be problematic because these instructions produce 16 bit values, you may need to add a movzx after such instructions to keep the JIT happy.

No, they just produce 64/32-bit values, but `crc32` has 16 bit operand,

Reply:
Sorry, only `crc32` needs  16 bit operands, not all.

Reply:
> Hmm, just because 16 bit versions of these instructions exist? 

Not really, `crc32 eax, ecx` produce different values from `crc32 eax, cx`, please look at the test cases.

Reply:
Yeah, crc32 is fine, the other 2 are problematic. It's unfortunate that they return the value in a 16 bit register.

Reply:
Oh, I see. Now, `EA_2BYTE` won't flow into other 2's codegen. Shall I check or assert here?

Reply:
It looks like it is OK as is. I see that in code gen you use the `emitTypeSize(targetType)` and since the target type is always TYP_INT/LONG it's fine. It would have been a problem if you used the operand type, that may be a small type.